routes/ai_brain.py

```diff
@@ -0,0 +1,77 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from typing import List
+from .. import models, schemas
+from fastapi.responses import FileResponse
+from ..database import get_db  
+
+router = APIRouter(prefix="/api/knowledge-base", tags=["Knowledge Base"])
+
+
+# LIST ALL ENTRIES
+@router.get("/", response_model=List[schemas.KnowledgeBaseOut])
+def list_all_entries(db: Session = Depends(get_db)):
+    return db.query(models.KnowledgeBase).all()
+
+
+# GET SINGLE ENTRY
+@router.get("/{entry_id}", response_model=schemas.KnowledgeBaseOut)
+def get_entry(entry_id: int, db: Session = Depends(get_db)):
+    entry = db.query(models.KnowledgeBase).filter(models.KnowledgeBase.id == entry_id).first()
+    if not entry:
+        raise HTTPException(status_code=404, detail="Knowledge base entry not found")
+    return entry
+
+
+# LIST ENTRIES FOR A USER
+@router.get("/users/{user_id}", response_model=List[schemas.KnowledgeBaseOut])
+def get_user_entries(user_id: int, db: Session = Depends(get_db)):
+    return db.query(models.KnowledgeBase).filter(models.KnowledgeBase.user_id == user_id).all()
```

Comment:
no logic for this specific three api's get all than get by user and tahn a single.

When we are getting all the knowledge base from user that will only give all the values no need for getting single knowedge base neither it is needed in fronend . 

No need of get all api as well